substrate/frame/revive/src/address.rs

```diff
@@ -124,24 +126,19 @@ where
 	fn map(account_id: &T::AccountId) -> DispatchResult {
 		ensure!(!Self::is_mapped(account_id), <Error<T>>::AccountAlreadyMapped);
```

Comment:
In the rare case that someone with an account that ends in `[0xEE; 12]` wants to legitimately call the `map` function, shouldn't it just silently pass instead of causing an error here?

Reply:
Maybe better to fail no, so that you can dry run the extrinsic (if polkadot wallet let you do that) realize that it fails and avoid paying the fees for a tx you don't need to submit

Reply:
Then some frontend app would need to deal with this edge case. Not sure whether this would be just one central app or could theoretically be any dapp frontend (yeah, probably not, if we are talking about native Ethereum dapps who don't know anything about H256 addresses)?

Reply:
Its a good point. If this is an edge case that needs to be handled depends on how wallets will determine if an address is already mapped.

I think we should make use of those new view functions and add a `mapped_address` which returns the `Eth` address for an AccountId32 if it is mapped. We should avoid making them doing the derivation themselves and then checking storage. Because then they would need to handle the edge case of 0xEE addresses. Keep in mind that they all be using SR25519 where 0xEE addresses are invalid.

So we should keep the error:
- In practice this edge case will hot happen because of wallets using sr25519
- Wallets will be using a view function to get the mapped address which will return the mapped address without registering first in case it ends in 0xEE